### PR TITLE
language: add internal command for generating source from dalf.

### PR DESCRIPTION
**Pull request by robin-da**
We add an internal command for generating DAML source code from .dalf
packages. Also adds an internal flag to tell the compiler, that it is
compiling generated source. We will be using this mainly for testing purposes.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/lib/DA/Cli/Damlc.hs [hunk 1]
```diff
@@ -208,7 +209,7 @@ cmdInspect =
     detailOpt =
         fmap (maybe DA.Pretty.prettyNormal DA.Pretty.PrettyLevel) $
             optional $ option auto $ long "detail" <> metavar "LEVEL" <> help "Detail level of the pretty printed output (default: 0)"
-    cmd = execInspect <$> inputFileOpt <*> outputFileOpt <*> jsonOpt <*> detailOpt
+    cmd = execInspect <$> inputDalfOpt <*> outputFileOpt <*> jsonOpt <*> detailOpt
```
Comment (cocreature) [inline compiler/damlc/lib/DA/Cli/Damlc.hs]: `inputDalfOpt` isn’t accurate. `inspect` can also take a `.dar` file and will the inspect the main dalf.

File: compiler/damlc/lib/DA/Cli/Damlc.hs [hunk 2]
```diff
@@ -303,6 +304,14 @@ cmdMergeDars =
   where
     cmd = execMergeDars <$> inputDarOpt <*> inputDarOpt <*> targetFileNameOpt
 
+cmdGenerateSrc :: Mod CommandFields Command
+cmdGenerateSrc =
+    command "generate-src" $
+    info (helper <*> cmd) $
+    progDesc "Generate DAML source code from a dalf package" <> fullDesc
+  where
+    cmd = execGenerateSrc <$> inputDalfOpt
```
Comment (cocreature) [inline compiler/damlc/lib/DA/Cli/Damlc.hs]: Should we add something like `-o output_directory`?

File: compiler/damlc/lib/DA/Cli/Damlc.hs [hunk 3]
```diff
@@ -913,6 +922,27 @@ execMergeDars darFp1 darFp2 mbOutFp =
...
+                        generateSrcPkgFromLf
+                            pkgId
+                            (MS.singleton
+                                 (stringToUnitId $ takeBaseName dalfFp)
+                                 pkgId)
+                            pkg
+                forM_ genSrcs $ \(path, src) -> do
+                    let fp = fromNormalizedFilePath path
+                    createDirectoryIfMissing True $ takeDirectory fp
+                    writeFile fp src
```
Comment (cocreature) [inline compiler/damlc/lib/DA/Cli/Damlc.hs]:
```suggestion
                    writeFileUTF8 fp src
```

File: compiler/damlc/lib/DA/Cli/Damlc.hs (hunk 2 above)
Comment (robin-da) [reply compiler/damlc/lib/DA/Cli/Damlc.hs]: let's keep it simple for now, I expect you create directory where you put your dalf into and run this.

File: compiler/damlc/lib/DA/Cli/Damlc.hs (hunk 1 above)
Comment (robin-da) [reply compiler/damlc/lib/DA/Cli/Damlc.hs]: hmm, before it was asking for a DAML file, which is completely off...

**Pull request merged**